README.md

```diff
@@ -67,6 +67,7 @@ Afterwards, any targets defined in the dependency can be used directly.
 CPMAddPackage(
   NAME          # The unique name of the dependency (should be the exported target's name)
   VERSION       # The minimum version of the dependency (optional, defaults to 0)
+  PATCHES       # Patch files to be applied sequentially using patch and PATCH_OPTIONS (optional)
```

Comment:
Would it make sense to provide more details or an example how path files are supplied somewhere in the Readme?

Reply:
@TheLartians YES indeed it makes sense to add a example! So I added this [example](/ScottBailey/CPM.cmake/tree/keyword_PATCHES/examples/highway) and a link in the [README](/ScottBailey/CPM.cmake/blob/9347302ec99e68171a64210e412e4bea9cdd3741/README.md?plain=1#L82). It's  very simple, but it's a real world use case. 

I looked back at the `cpm_add_patches` and felt like a more verbose description was in order, so I [improved the text](https://github.com/ScottBailey/CPM.cmake/blob/9347302ec99e68171a64210e412e4bea9cdd3741/cmake/CPM.cmake#L467-L469). I also made a couple attempts to add an output variable and none of the paths I chose felt right to me, so I'd rather just merge as is. I'm open to more guidance here, but my free time will be greatly reduced soon...

Reply:
I thought about next: should we document/write an example somewhere about PATCH_COMMAND ? Since maybe someone wants to make a patch for an old version of a package, then it happens that in a new version it's applicable but with a shift. AFAIK, there's some options in `patch` command for this like `-p` and `-F` for such cases, and PATCHES usually goes with default options .

Reply:
> I thought about next: should we document/write an example somewhere about PATCH_COMMAND ? 

This wouldn't necessarily be a bad idea. It was a non-zero effort for me to figure out how to patch things and it was tedious in my CMake files. Which is my motivation to write this PR.

I think a future PR to add modifiers to this `PATCHES` command for any CLI options that are commonly used is the way to go.

Reply:
As illustration, here is a **relatively simple** patch command:
`PATCH_COMMAND "${pexe}\\\;-p\\\;<\\\;${CMAKE_CURRENT_LIST_DIR}/000-srclib-2.patch\\\;&&\\\;${pexe}\\\;-p\\\;<\\\;${CMAKE_CURRENT_LIST_DIR}/000-srclib-3.patch"`

I don't really want to try to write the clear and complete documentation for that! I'd rather write an interface to simplify it to something much more intuitive.